### DISCUSSION TITLE
Different migrations for different DB types?

### DISCUSSION THREAD

**Discussion by duckcantcode** [diesel-rs/diesel#4403] [Q&A] [answered]
I'm writing a program in Rust that I want to be able to use either SQLite *or* PostgreSQL. Right now, it only works with SQLite and it runs a "schema" every startup that has a bunch of `CREATE TABLE IF NOT EXISTS`. How does this translate to diesel, is this the same as a migration? And do I need to have two separate migration folders, one for SQLite and another for Postgre?

**Comment by weiznich** [comment, marked as answer]
> How does this translate to diesel, is this the same as a migration? 

It's up to you how to translate this to diesel. You don't need to use migrations for that, you could also just execute these queries on startup manually like you've done before. That said: Migrations are likely a good solution for that, you would put your schema setup into a migration and diesel would take care to run this migration if it was not run previously.

> And do I need to have two separate migration folders, one for SQLite and another for Postgre?

Yes you would need to have a separate migration folder for each backend
